rust/src/wrapper/device/mod.rs

```diff
@@ -295,6 +461,15 @@ impl Device {
         })?;
         Ok(())
     }
+
+    /// Gets the Device's `random_address` property
+    pub fn random_address(&self) -> PyResult<Address> {
+        Python::with_gil(|py| {
+            self.obj
+                .getattr(py, intern!(py, "random_address"))
```

Comment:
Is this actually random? From some brief but possibly flawed perusal of the code, it looks like this would end up as all zeros?

Reply:
`random_address` is set to random bytes retrieved from the Controller during `power_on`
```
if self.le_enabled:
    # Set the controller address
    if self.random_address == Address.ANY_RANDOM:
        # Try to use an address generated at random by the controller
        if self.host.supports_command(HCI_LE_RAND_COMMAND):
            # Get 8 random bytes
            response = await self.send_command(
                HCI_LE_Rand_Command(), check_result=True  # type: ignore[call-arg]
            )

            # Ensure the address bytes can be a static random address
            address_bytes = response.return_parameters.random_number[
                :5
            ] + bytes([response.return_parameters.random_number[5] | 0xC0])

            # Create a static random address from the random bytes
            self.random_address = Address(address_bytes)
```